lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +246,30 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding
+    !@default_encoding.nil? || (@default_encoding.nil? && !@columns_to_encode.empty?)
+  end
+
+  # make sure the encoding is uniform over fields
+  def convert(key, value)
+    return value unless should_convert(key, value)
+    converter = find_converter_for(key, value)
+    converter.convert(value)
+  end
+
+  def should_convert(key, value)
+    column_keys = @columns_to_encode.keys
```

Comment:
extracting the keys at each columns for each row seems inefficient, the `column_keys` can be cached no?